lib/widgets/message_list.dart

```diff
@@ -161,7 +189,26 @@ class _MessageListState extends State<MessageList> {
           child: Center(
             child: ConstrainedBox(
               constraints: const BoxConstraints(maxWidth: 760),
-              child: _buildListView(context))))));
+              child: NotificationListener<ScrollEndNotification>(
+                onNotification: (scrollEnd) {
+                  final metrics = scrollEnd.metrics;
+                  if (metrics.atEdge && metrics.pixels == 0) {
+                    _scrollToBottomVisibleValue.value = false;
+                  } else {
+                    _scrollToBottomVisibleValue.value = true;
+                  }
+                  return true;
+                },
+                child: Stack(
+                  children: <Widget>[
+                    _buildListView(context),
+                    Container(
+                      alignment: Alignment.bottomRight,
+                      child: ScrollToBottomButton(scrollController: scrollController, visibleValue: _scrollToBottomVisibleValue),
```

Comment:
Is there a reason to prefer `Container` to control this alignment over `Positioned`? The latter seems to be what `Stack`'s docs suggest, but I haven't dug into what consequences the difference might have.

Reply:
I looked over the flutter code for the `Stack` behaviour (https://github.com/flutter/flutter/blob/305c36f3d2380b7bd69e1829b7f7d4dcdb157928/packages/flutter/lib/src/rendering/stack.dart#L304) and it appears the difference is that non-positioned elements (not using `top`,`left`,etc) are considered in the calculation of the `Stack` size. So I believe if the list got smaller than the button the stack would expand to show the button. I don't think this is a necessary edge case to consider, and switching to using a `Positioned` element makes sense as it seems more inline with what the documentation for `Stack` suggests

Reply:
Cool. In this case I think the other element in the Stack is always expanding to fill the max size its parent allows, so the only way it'd be smaller than the button anyway is if the constraints force it that small, and keeping the button non-positioned wouldn't affect the size anyway.

In any event, I agree that's not an edge case we need to worry about — there's no way our UI can work out at a size that small.